### PR TITLE
[server][dvc] Fix flaky integration test BlobP2PTransferAmongServersTest

### PR DESCRIPTION
**Pull request by jingy-li**
## [server][dvc] Fix flaky integration test BlobP2PTransferAmongServersTest

This PR fixes the following:

1. Fix the flaky integration test.

## How was this PR tested?
integration test. 

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by huangminchn — APPROVED** [review]
Thanks a lot Jingyan!

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public void startConsumption(VeniceStoreVersionConfig storeConfig, int partition
           storeAndVersion.getFirst(),
           storeAndVersion.getSecond().getNumber(),
           partition,
-          serverConfig.getBlobTransferDisabledOffsetLagThreshold());
+          storeConfig.getBlobTransferDisabledOffsetLagThreshold());
```
Comment (sushantmane) [inline clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java:89]: Any reason for this change?

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/BlobP2PTransferAmongServersTest.java [hunk 2]
```diff
@@ -176,8 +176,8 @@ public void testBlobTransferThrowExceptionIfSnapshotNotExisted() throws Exceptio
       }
     }
 
+    cluster.stopAndRestartVeniceServer(server1.getPort());
```
Comment (sushantmane) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/BlobP2PTransferAmongServersTest.java:179]: Can we add assertion, if possible, to see calls were made to P2P service?

File: clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java (hunk 1 above)
Comment (sushantmane) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java:89]: Also can we rename serverConfig method name to `getServerLevelBlobTransferDisabledOffsetLagThreshold` or something?

Comment (sixpluszero) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java:89]: Synced offline with Jingyan, we don't need this change. We need to add to global-config for DVC to pick up override.

Comment (jingy-li) [reply clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java:89]: This issue is related to the deployment not successfully picking up the getBlobTransferDisabledOffsetLagThreshold config (unrelated to the fix for the integration test). The root cause is that the global-config was not added, not related to the code issue. I will revert this change.

**Review by sushantmane — APPROVED** [review]
LGTM! Thanks, @jingy-li!

**Pull request merged**